client/customer-effort-score-tracks/customer-effort-score-tracks-container-analytics.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { compose } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+import PropTypes from 'prop-types';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import CustomerEffortScoreTracks from './customer-effort-score-tracks';
+
+const SHOWN_FOR_ACTIONS_OPTION_NAME = 'woocommerce_ces_shown_for_actions';
+const ACTION_NAME = 'woocommerce.admin.analytics.filtered';
+
+/**
+ * Listens for ACTION_NAME event from the client/analytics/components/report-filters/index.js
+ * and renders CustomerEffortScoreTracks conditionally.
+ 
+ * @param {Object}   props           Component props.
+ * @param {Function} props.getOption Function to get WordPress options.
+ */
+function CustomerEffortScoreTracksContainerAnalytics( { getOption } ) {
+	// Save the result of hasShown() so that the subsequent
+	// woocommerce.admin.analytics.filtered actions do not trigger the survey again
+	// on the same page
+	let hasShownResultCache = false;
+
+	// We don't want to render CustomerEffortScoreTracks for the first page load
+	const [ showComponent, setShowComponent ] = useState( false );
+
+	const hasShown = () => {
+		const shownActions = getOption( SHOWN_FOR_ACTIONS_OPTION_NAME ) || [];
+		if ( shownActions.indexOf( ACTION_NAME ) !== -1 ) {
+			return true;
+		}
+
+		return false;
+	};
+
+	const onAnalyticFilterChanged = () => {
+		if ( hasShownResultCache === true ) {
+			return;
+		}
+
+		if ( hasShown( ACTION_NAME ) === false ) {
+			setShowComponent( true );
+			hasShownResultCache = true;
+		}
+	};
+
+	// Listen for the action
+	wp.hooks.addAction(
+		ACTION_NAME,
+		'woocommerce.admin',
+		onAnalyticFilterChanged
+	);
+
+	function getComponent() {
+		return (
+			<CustomerEffortScoreTracks
+				key={ 999 }
+				action={ ACTION_NAME }
+				label={ __(
+					'How easy was it to filter your store analytics?',
+					'woocommerce-admin'
+				) }
+				onSubmitLabel={ __(
+					'Thank you for your feedback!',
+					'woocommerce-admin'
+				) }
+				trackProps={ {} }
+			/>
+		);
+	}
+
+	return showComponent ? getComponent() : null;
+}
+
+CustomerEffortScoreTracksContainerAnalytics.propTypes = {
+	/**
+	 * Function to get WordPress options.
+	 */
+	getOption: PropTypes.func,
+};
+
+export default compose(
+	withSelect( ( select ) => {
+		const { getOption } = select( OPTIONS_STORE_NAME );
+		return { getOption };
```

Comment:
Thank you for the link. I updated it with the latest commit.

Reply:
That's a good idea. Thank you! With the change, I also removed `hasShown()` method as it is unnecessary. The function is a bit cleaner and simpler now :) https://github.com/woocommerce/woocommerce-admin/blob/e7b69e97850ab1230f3882e32c6a59ef4d9dd356/client/customer-effort-score-tracks/customer-effort-score-tracks-container-analytics.js#L36

Reply:
@jeffstieler I might be overthinking, but calling `getOption` function within `withSelect` function might be making an unnecessary call (option rest API) to the backend as `CustomerEffortScoreTracksContainerAnalytics` component renders nothing initially. I think it might be better to lazy call `getOption` like the original commit. What do you think?

Reply:
You are not at all overthinking - that's a nice catch. In other parts of the app we use the [`withOptionsHydration()`](https://github.com/woocommerce/woocommerce-admin/blob/a9acc231e78640a7adf715d486b5780e3c0f39d5/packages/data/src/options/with-options-hydration.js#L12) higher order component to hydrate preloaded options (typically from a global variable) like in [`client/layout/index.js`](https://github.com/woocommerce/woocommerce-admin/blob/d6c80bb709197ea8a4ddd87f517cbde2caca970b/client/layout/index.js#L226-L232).

If you choose to use the hydration approach, you'll need to add the `woocommerce_ces_shown_for_actions` options to the the preloaded options array using the `woocommerce_admin_preload_options` filter. Here's [an example](https://github.com/woocommerce/woocommerce-admin/blob/e89cc8a2ef9d193a376c0e5fb8cf6376e7c86b06/src/Features/Homescreen.php#L46) from the Homescreen feature class.

Reply:
@jeffstieler Thank you for the references. I will look into them.